File: 0485-max-consecutive-ones/485.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        curr_count, ans = 0, 0
        
        for num in nums:
            if num:
                curr_count += 1
                if curr_count > ans:
                    ans = curr_count
            else:
                curr_count = 0
        return ans

File: 0485-max-consecutive-ones/485.rs
use std::cmp::max;

impl Solution {
    pub fn find_max_consecutive_ones(nums: Vec<i32>) -> i32 {
        let (mut res, mut curr_count) = (0, 0);
        for &num in &nums {
            if num == 1 {
                curr_count += 1;
                res = max(curr_count, res);
            } else {
                curr_count = 0;
            }
        }
        res
    }
}
